### PR TITLE
New reverse proxy handling, use `RewriteRule` instead of `ProxyPass`

### PR DESCRIPTION
**Pull request by federicoballarini**
Webtop `5.10.0` introduce a new auth log where are recorded all the access attempt and the associated information, like sessions, location (if possible) and source IP address.
To determine the source IP address, WebTop inspects the header `X-Forwarded-For` if it is present, otherwise falls back on the request IP address.

Due to a bug in the Centos provided Apache HTTP version `2.4.6`,  if `ProxyPass` is used to forward the request to WebTop, after some time it stops adding the X-Forwarded-For into the requests. As a consequence of this bug, Webtop starts to report `127.0.0.1` as the IP source of the authentication attempts.

A workaround for this bug is:

* Rewrite the proxy directive using `RewriteRule` with the `P` flag instead of `ProxyPass`
* Move part of the proxy configuration in a `.htaccess` file

NethServer/dev#6463

### CODE REVIEW

**Review by Amygos — CHANGES_REQUESTED** [review]
You have to add the path  `/var/www/html/webtop/.htaccess` into the list of templates to expand on package update, in the `createlinks` file.

**Comment by federicoballarini** [comment]
> You have to add the path `/var/www/html/webtop/.htaccess` into the list of templates to expand on package update, in the `createlinks` file.

Yes, you are right. My bad. I missed to report it from dev... :/

**Comment by Amygos** [comment]
The PR seems ok for me, but before the merge, we need to out a better explanation on why we need it.

File: root/etc/e-smith/templates/var/www/html/webtop/.htaccess/10base [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+#
+
+RewriteEngine on
+RewriteCond %\{HTTP:Upgrade\} websocket [NC]
+RewriteCond %\{HTTP:Connection\} upgrade [NC]
+RewriteRule .* ws://127.0.0.1:58080/webtop/push%\{REQUEST_URI\} [P,L]
+
+RewriteEngine on
+RewriteCond %\{REQUEST_URI\} !/webtop/index.*
+RewriteRule .* http://127.0.0.1:58080%\{REQUEST_URI\} [P,L]
```
Comment (gsanchietti) [inline root/etc/e-smith/templates/var/www/html/webtop/.htaccess/10base:12]: This fragment doesn't contain any variable, why is it a template?

Please convert it to a simple static file.

**Review by gsanchietti — CHANGES_REQUESTED** [review]

File: root/etc/e-smith/templates/var/www/html/webtop/.htaccess/10base (hunk 1 above)
Comment (federicoballarini) [reply root/etc/e-smith/templates/var/www/html/webtop/.htaccess/10base:12]: > This fragment doesn't contain any variable, why is it a template?
> 
> Please convert it to a simple static file.

I need to remove also /var/www/html/.htaccess from createlinks, correct? If an user made a change on .htaccess file, the event nethserver-webtop5-update will not restore it to default. Is this the correct behaviour?

Comment (gsanchietti) [reply root/etc/e-smith/templates/var/www/html/webtop/.htaccess/10base:12]: Yes, remove the entry also from createlinks.
This is not a customizable file: if you change it, WebTop will stop working.
The user should not modify it just like any other file belonging to a RPM and not marked as config file.

File: nethserver-webtop5.spec [hunk 2]
```diff
@@ -45,6 +45,8 @@ mkdir -p root/usr/share/webtop/updates/post/main
 
 mkdir -p root/etc/webtop/webtop
 
+mkdir -p root/var/www/html/webtop
+
```
Comment (gsanchietti) [inline nethserver-webtop5.spec:49]: You do not this anyore
```suggestion
```

**Review by gsanchietti — CHANGES_REQUESTED** [review]

File: createlinks [hunk 3]
```diff
@@ -40,7 +40,7 @@ event_templates($event,
                      '/etc/dovecot/dovecot.conf',
                      '/etc/httpd/conf.d/webtop.conf',
                      '/etc/httpd/conf.d/default-virtualhost.inc',
-		     '/etc/webtop/webtop.properties',
+                     '/etc/webtop/webtop.properties',
```
[the remark this answers is not in the document]
Comment (federicoballarini) [reply createlinks:43]: Ok... :( :D

[the remark this answers is not in the document]
Comment (gsanchietti) [reply createlinks:43]: Just open another PR with this ;)

File: nethserver-webtop5.spec [hunk 4]
```diff
@@ -300,4 +300,3 @@ cp -a api/* %{buildroot}/usr/libexec/nethserver/api/%{name}/
 
 * Wed Mar 08 2017 Giacomo Sanchietti <giacomo.sanchietti@nethesis.it> - 1.0.0-1
 - WebTop 5 - NethServer/dev#5225
```
Comment (gsanchietti) [inline nethserver-webtop5.spec:302]:
```suggestion
- WebTop 5 - NethServer/dev#5225

```

File: nethserver-webtop5.spec [hunk 5]
```diff
@@ -300,4 +300,3 @@ cp -a api/* %{buildroot}/usr/libexec/nethserver/api/%{name}/
...
-
```
Comment (gsanchietti) [inline nethserver-webtop5.spec:303]: Please, avoid also this change

**Comment by Amygos** [comment]
I have updated the PR's description with the full background explanation.

File: nethserver-webtop5.spec (hunk 5 above)
Comment (federicoballarini) [reply nethserver-webtop5.spec:303]: Ops... this is clearly an error and I can't remember when I changed it...

File: createlinks (hunk 3 above)
[the remark this answers is not in the document]
Comment (federicoballarini) [reply createlinks]: After merging this I will open it :)

**Review by gsanchietti — APPROVED** [review]
:heavy_check_mark: 

@Amygos you can squash&merge whenever you prefer

**Review by Amygos — APPROVED** [review]

**Pull request merged**